sdk/cosmos/azure-cosmos-spark_3-3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark
+
+import java.util
+// scalastyle:off underscore.import
+// scalastyle:on underscore.import
+import org.apache.spark.sql.catalyst.analysis.{NamespaceAlreadyExistsException, NoSuchNamespaceException}
+import org.apache.spark.sql.connector.catalog.{NamespaceChange, SupportsNamespaces}
+
+// scalastyle:off underscore.import
+
+class CosmosCatalog
+  extends CosmosCatalogBase
+    with SupportsNamespaces {
+
+    override def listNamespaces(): Array[Array[String]] = {
+        super.listNamespacesBase()
+    }
+
+    @throws(classOf[NoSuchNamespaceException])
+    override def listNamespaces(namespace: Array[String]): Array[Array[String]] = {
+        super.listNamespacesBase(namespace)
+    }
+
+    @throws(classOf[NoSuchNamespaceException])
+    override def loadNamespaceMetadata(namespace: Array[String]): util.Map[String, String] = {
+        super.loadNamespaceMetadataBase(namespace)
+    }
+
+    @throws(classOf[NamespaceAlreadyExistsException])
+    override def createNamespace(namespace: Array[String],
+                                     metadata: util.Map[String, String]): Unit = {
+        super.createNamespaceBase(namespace, metadata)
+    }
+
+    @throws(classOf[UnsupportedOperationException])
+    override def alterNamespace(namespace: Array[String],
+                                changes: NamespaceChange*): Unit = {
+        super.alterNamespaceBase(namespace, changes)
+    }
+
+    @throws(classOf[NoSuchNamespaceException])
+    override def dropNamespace(namespace: Array[String], cascade: Boolean): Boolean = {
+        super.dropNamespaceBase(namespace)
```

Comment:
This is functionally incorrect. You ignore the cascade flag. Our implementation always assumed cascading delete (so automatically deleted Coamos Database - which service-side will result in deleting containers). In Saprk 3.3 this should change - only if cascade==true we should call the base method. Otherwise throw an exception (not sure which one, needs some digging in Spark 3.3 spec)

This is blocking merge

Reply:
Got it - so we should be if !cascade -> then check whether there is any containers under the database, if there is any, then throw some notEmpty exception, else just go ahead delete the database right?

Reply:
LGTM